reactioncommerce/reaction
# [3.0.0] Consolidate translation plugins

Impact: **minor**  
Type: **refactor**

## Changes
- Move `address` from `plugins` to `core-services` because it is core
- Several plugins were carried over from 2.x but now contain only translations. This merges all translations from those plugins into a single new plugin named `reaction-translations`. These translations are used by the Reaction Admin project. The following plugins are thus deleted:
    - src/core-services/core
    - src/plugins/checkout
    - src/plugins/dashboard
    - src/plugins/product-admin
    - src/plugins/product-variant
    - src/plugins/templates
    - src/plugins/ui

The templates plugin also defined the `Templates` MongoDB collection, but since that collection is only used by the `email-templates` plugin, I moved it there.

## Breaking changes
Some plugins are removed but since they contained only translations and those translations are still present in the new plugin, it's unlikely this will break anything.

## Testing
- Verify the app starts.
- Verify http://localhost:3000/locales/resources.json?lng=en&ns=reaction-translations returns the English translations from the new plugin.
- I wrote a script to move/combine all the translations files, so a spot check of a couple files should be sufficient to verify the script worked correctly.
